packages/mobile/src/screens/track-screen/TrackRemixesScreen.tsx

```diff
@@ -56,28 +56,36 @@ export const TrackRemixesScreen = () => {
   const navigation = useNavigation()
   const lineup = useSelector(getRemixesTracksLineup)
   const count = useSelector(getCount)
-  const track = useSelector(getTrack)
-  const user = useSelector(getUser)
   const dispatch = useDispatch()
-  const trackId = track?.track_id
-
   const styles = useStyles()
   const { params } = useRoute<'TrackRemixes'>()
 
-  useEffectOnce(() => {
-    dispatch(fetchTrack(params))
+  const hasId = 'id' in params
+  const hasPermalink = 'handle' in params && 'slug' in params
+
+  const { data: trackById } = useTrack(hasId ? params.id : null, {
+    enabled: hasId
   })
 
+  const { data: trackByPermalink } = useTrackByPermalink(
+    hasPermalink ? `/${params.handle}/${params.slug}` : null,
+    { enabled: hasPermalink }
+  )
+
+  const track = trackById ?? trackByPermalink
+  const trackId = track?.track_id
+  const user = track?.user
+
   useEffect(() => {
     if (trackId) {
+      dispatch(fetchTrackSucceeded({ trackId }))
```

Comment:
does anything actually listen for this?

Reply:
yeah actually. there is still a track saga that does a few different things, like adding it to a lineup. That used to be in the "fetchTrack" saga, so now i moved the fetch part to the hook and have a "suceeded" call that does the post fetch stuff.